### ISSUE TITLE
Add ability to hide some apps from the app drawer

### ISSUE THREAD

**Issue by Lonami** [stanipintjuk/Silverfish#10]
Most launchers don't show themselves in the application drawer (it's kind of pointless). Should Silverfish Launher hide the main activity from the launcher too? Perhaps, in a future, it could show a "Launcher settings" icon instead (with the main application icon and perhaps a settings wheel).

**Comment by stanipintjuk** [comment]
I agree, it is kind of annoying

**Comment by neckcen** [comment]
It would be useful to be able to hide any app (with the launcher being hidden by default).

Use case: I manage the phones for some of my less tech-literate family members and there are some apps they just don't need to see/use.

**Comment by Lonami** [comment]
That's a good idea too!

**Comment by Teyro** [comment]
Will be nice, i use Silverfish launcher for testing on my preschool / kindergarden tablet at work...It is nice and simple and 5-6 years old children can use it without problems.... But i have to Hide Settings and so on........ And i have created a adults categorey ....But some Parents thougt that is something with p**n... So hideing apps would be nice...

**Comment by stanipintjuk** [comment]
@Teyro 

> And i have created a adults categorey ....But some Parents thougt that is something with p**n...

Lol!

I will add this feature as soon as I have added all the basic features I have intended for silverifish, and fixed most of the bugs (i.e. "some time after version 1.0"), because I would have to create a different database schema... and it scares me :(

**Comment by Lonami** [comment]
Different database schema? Wouldn't it be enough with adding a `boolean hidden`?

**Comment by stanipintjuk** [comment]
Yes, but we also have to write a function to update the old schema to the new schema without wiping out existing entries.... And also while I do that I might as well think through the current schema and make it a little bit more sane

**Comment by Lonami** [comment]
> but we also have to write a function to update the old schema to the new schema without wiping out existing entries

Don't bother writing a function to update the old schema to the new one. This launcher is an alpha. People should know the risk when using alpha versions.

> [...] current schema and make it a little bit more sane

That's always good though :) It's hard to think a design which is easily expandable from the beginning, at least I need to learn from my mistakes before I get there :P
